### PR TITLE
Reduce lenght of attribute InvTypeCode to 1-8 characters

### PR DESCRIPTION
**Pull request by BraunLukas**
Commit changes:
- add simpleType "def_invTypeCode_string" to alpinebits.xsd (string 1 to 8 char)
- change attribute type of OTA_HotelAvailNotifRQ->AvailStatusMessage->InvTypeCode to "def_invTypeCode_string"
- change attribute type of OTA_ResRetrieveRS->ReservationList->HotelReservation->RoomStay->RoomType->RoomTypeCode to "def_invTypeCode_string"
- change attribute type of OTA_HotelInvNotifRQ->SellableProduct->InvTypeCode to "def_invTypeCode_string"
- change attribute type of OTA_HotelRatePlanNotifRQ->RatePlan->Rate->InvTypeCode to "def_invTypeCode_string"

### CODE REVIEW

File: files/schema-xsd/alpinebits.xsd [hunk 1]
```diff
@@ -302,7 +302,7 @@
 
                                         <xs:element name="RoomType">
                                           <xs:complexType>
-                                            <xs:attribute name="RoomTypeCode"           type="def_nonempty_string"/>
+                                            <xs:attribute name="RoomTypeCode"           type="def_invTypeCode_string"/>
```
Comment (BraunLukas) [inline files/schema-xsd/alpinebits.xsd]: Not sure about this one...


... this RoomTypeCode attribute should be identical to the InvTypeCode attribute used for room categories (see section 4.1.1)

Comment (danielegobbetti) [inline files/schema-xsd/alpinebits.xsd]: I agree that we should ensure that the codes used for cross-reference should be coherent, as for the best way to achieve this I'd wait for @chris-mair opinion .

Comment (rambinho) [inline files/schema-xsd/alpinebits.xsd]: Daniele will change this for Inventory, @chris-mair could you do a quick cross check if lenght 8 is ok the other messages?

File: files/schema-xsd/alpinebits.xsd [hunk 2]
```diff
@@ -1981,6 +1981,15 @@
     </xs:restriction>
   </xs:simpleType>
 
+  <!-- String type defining Rooms InvTypeCode  -->
+
+  <xs:simpleType name="def_invTypeCode_string">
+    <xs:restriction base="xs:string">
+      <xs:minLength value="1"/>
+      <xs:maxLength value="16"/>
```
Comment (BraunLukas) [inline files/schema-xsd/alpinebits.xsd]: Change this value to 8 before merging...

**Comment by chris-mair** [comment]
Hi,

currently we have used non empty string al lot:

earth:files chris$ grep def_nonempty_string schema-xsd/alpinebits.xsd | wc -l
      58

many of these are for OTA IDs and Codes:

earth:files chris$ grep def_nonempty_string schema-xsd/alpinebits.xsd | egrep -i '(code|id)'  | wc -l
      29

Most of these have some sort of limits and restrictions - generally different.

I'm reluctant to just change it for the invtype.

I suggest we open a topic for the next release where if this is deemed important we can
make a major update to the schema file introducing the correct limits and restrictions to ALL
our Codes and IDs, following OTA.

I will reject this pull request now.

Please post a request (to update all Codes and IDs) to the AlpineBits forum after 2015-07 is released.

-- Chris.

**Comment by chris-mair** [comment]
closed as per my last comment

**Comment by chris-mair** [comment]
reopened for merge into 2015-07b

**Comment by chris-mair** [comment]
OK. I've manually merged your branch (updating it for the new inventory and adding BookingRule -> Code.

Thanks, and sorry it took a year :)

-- Chris

**Pull request merged**